File: JAVA/2337. Move Pieces to Obtain a String.java
class Solution {
    public boolean canChange(String s, String t) {
        int i=0,j=0;
        while(i<s.length()&&j<t.length()){
            while(i<s.length()&&s.charAt(i)=='_')
            i++;
            while(j<t.length()&&t.charAt(j)=='_')
            j++;
            if(i==s.length()||j==s.length())
            return i==s.length()&&j==s.length();
            if((s.charAt(i)!=t.charAt(j))||(s.charAt(i)=='L'&&i<j)||(s.charAt(i)=='R'&& i>j))
            return false;
            i++;
            j++;
        }
        return true;
    }
}

File: C++/2337. Move Pieces to Obtain a String.cpp
class Solution {
public:
    bool canChange(string s, string t) {
        int i=0,j=0;
        while(i<s.size()||j<s.size()){
            while(i<s.size()&&s[i]=='_')
            i++;
            while(j<s.size()&&t[j]=='_')
            j++;
            if(i==s.size()||j==s.size())
            return i==s.size()&&j==s.size();
            if((s[i]!=t[j])||(s[i]=='L'&&i<j)||(s[i]=='R'&&i>j))
            return false;
            i++;
            j++;
        }
        return true;
    }
};